test/src/test/java/org/corfudb/integration/LogReplicationDataReceiverIT.java

```diff
@@ -0,0 +1,305 @@
+            CorfuRecord<TableDescriptors, TableMetadata> corfuRecord =
+                    standbyRuntime.getTableRegistry().getRegistryTable().get(entry.getKey());
+            Assert.assertFalse(corfuRecord.getMetadata().getTableOptions().getIsFederated());
+            Assert.assertTrue(corfuRecord.getMetadata().getTableOptions().getRequiresBackupSupport());
+        }
+
+        verifyInLogEntrySyncState();
+    }
+
+    /**
+     * This test verifies entries for registry table are correctly applied during log entry sync.
+     */
+    @Test
+    @SuppressWarnings("checkstyle:magicnumber")
+    public void testRegistryTableInLogEntrySync() throws Exception {
+        setupActiveAndStandbyCorfu();
+        openLogReplicationStatusTable();
+
+        // Initial setup for streams to replicate for both active and standby side
+        int numTableActive = 4;
+        int numTableStandby = 4;
+        mapNameToMapActive = new HashMap<>();
+        mapNameToMapStandby = new HashMap<>();
+        setupStreamsToReplicate(numTableActive, true);
+        setupStreamsToReplicate(numTableStandby, false);
+
+        for (int i = 1; i <= numTableActive / 2; i++) {
+            String mapName = TABLE_PREFIX + i;
+            Table<StringKey, IntValueTag, Metadata> tableActive =
+                    corfuStoreActive.openTable(NAMESPACE, mapName, StringKey.class,
+                            IntValueTag.class, Metadata.class,
+                            TableOptions.fromProtoSchema(IntValueTag.class, TableOptions.builder().build()));
+            mapNameToMapActive.put(mapName, tableActive);
+
+            Table<StringKey, IntValueTag, Metadata> tableStandby =
+                    corfuStoreStandby.openTable(NAMESPACE, mapName, StringKey.class,
+                            IntValueTag.class, Metadata.class,
+                            TableOptions.fromProtoSchema(IntValueTag.class, TableOptions.builder().build()));
+            mapNameToMapStandby.put(mapName, tableStandby);
+        }
+
+
+        // Add Data for Snapshot Sync
+        writeToActive(0, NUM_WRITES);
+        // Confirm data does exist on Active Cluster
+        for(Table<StringKey, IntValueTag, Metadata> map : mapNameToMapActive.values()) {
+            assertThat(map.count()).isEqualTo(NUM_WRITES);
+        }
+
+        // Confirm data does not exist on Standby Cluster
+        for(Table<StringKey, IntValueTag, Metadata> map : mapNameToMapStandby.values()) {
+            assertThat(map.count()).isEqualTo(0);
+        }
+
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_ACTIVE;
+        startActiveLogReplicator();
+
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_STANDBY;
+        startStandbyLogReplicator();
+
+        verifyDataOnStandby(NUM_WRITES);
+
+        // Verify snapshot sync completed and cluster went into log entry sync
+        verifyInLogEntrySyncState();
+
+        stopActiveLogReplicator();
```

Comment:
why do we need to shut it down?

Reply:
to make sure Standby's registry table already has the entries before replication

Reply:
Yes I was meaning the registry table's records .. sorry for the confusion. Standby's records have `false` while Active's records have `true`, which is verified in line 228